src/OpenSage.Game/Logic/OrderGenerators/ConstructBuildingOrderGenerator.cs

```diff
@@ -62,15 +62,26 @@ public OrderGeneratorResult TryActivate(Scene3D scene)
                 }
             }
 
-            var order = Order.CreateBuildObject(scene.GetPlayerIndex(scene.LocalPlayer), _definitionIndex, _position, _angle);
+            var player = scene.LocalPlayer;
+            if (player.Money < _buildingDefinition.BuildCost)
+            {
+                return OrderGeneratorResult.Failure("Not enough cash");
+            }
+            else
+            {
+                player.Money -= (uint) _buildingDefinition.BuildCost;
```

Comment:
We should remove the money in the build order processor. If we spend any money in this function, it won't be networked correctly.